### PR TITLE
WFLY-15990 NON_XA transaction mode causes ClassCastException in WF preview

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-15990

### CODE REVIEW

File: ee-9/common/src/main/resources/modules/system/layers/base/org/jboss/as/clustering/infinispan/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2010, Red Hat, Inc., and individual contributors
```
Comment (scottmarlow) [inline ee-9/common/src/main/resources/modules/system/layers/base/org/jboss/as/clustering/infinispan/main/module.xml:5]: Copyright should be 2022

**Comment by scottmarlow** [comment]
I'll test this locally.

**Comment by pferraro** [comment]
Thanks @scottmarlow !

**Comment by scottmarlow** [comment]
@pferraro I'm now getting the following in server.log:

```
Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: CTS-EM2] Unable to build Hibernate SessionFactory
	at org.hibernate@5.3.24.Final//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.persistenceException(EntityManagerFactoryBuilderImpl.java:1327)
	at org.hibernate@5.3.24.Final//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1253)
	at org.hibernate.jipijapa-hibernate5-3@27.0.0.Beta1-SNAPSHOT//org.jboss.as.jpa.hibernate5.TwoPhaseBootstrapImpl.build(TwoPhaseBootstrapImpl.java:44)
	at org.jboss.as.jpa@27.0.0.Beta1-SNAPSHOT//org.jboss.as.jpa.service.PersistenceUnitServiceImpl$1$1.run(PersistenceUnitServiceImpl.java:171)
	... 10 more
Caused by: org.hibernate.service.spi.ServiceException: Unable to create requested service [org.hibernate.cache.spi.CacheImplementor]
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:275)
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:237)
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:214)
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.SessionFactoryServiceRegistryImpl.getService(SessionFactoryServiceRegistryImpl.java:109)
	at org.hibernate@5.3.24.Final//org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:239)
	at org.hibernate@5.3.24.Final//org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:467)
	at org.hibernate@5.3.24.Final//org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1250)
	... 12 more
Caused by: org.hibernate.cache.CacheException: HHH025009: Unable to start region factory
	at org.infinispan.hibernate-cache@13.0.6.Final//org.infinispan.hibernate.cache.v53.InfinispanRegionFactory.start(InfinispanRegionFactory.java:248)
	at org.hibernate@5.3.24.Final//org.hibernate.cache.internal.EnabledCaching.<init>(EnabledCaching.java:82)
	at org.hibernate@5.3.24.Final//org.hibernate.engine.spi.CacheInitiator.initiateService(CacheInitiator.java:33)
	at org.hibernate@5.3.24.Final//org.hibernate.engine.spi.CacheInitiator.initiateService(CacheInitiator.java:24)
	at org.hibernate@5.3.24.Final//org.hibernate.service.spi.SessionFactoryServiceInitiator.initiateService(SessionFactoryServiceInitiator.java:30)
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.SessionFactoryServiceRegistryImpl.initiateService(SessionFactoryServiceRegistryImpl.java:68)
	at org.hibernate@5.3.24.Final//org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:263)
	... 18 more
Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000282: Cannot enable Invocation Batching when the Transaction Mode is NON_TRANSACTIONAL, set the transaction mode to TRANSACTIONAL
	at org.infinispan@13.0.6.Final//org.infinispan.configuration.cache.InvocationBatchingConfigurationBuilder.validate(InvocationBatchingConfigurationBuilder.java:57)
	at org.infinispan@13.0.6.Final//org.infinispan.configuration.cache.ConfigurationBuilder.validate(ConfigurationBuilder.java:180)
	at org.infinispan@13.0.6.Final//org.infinispan.configuration.cache.ConfigurationBuilder.build(ConfigurationBuilder.java:236)
	at org.infinispan@13.0.6.Final//org.infinispan.configuration.cache.ConfigurationBuilder.build(ConfigurationBuilder.java:226)
	at org.infinispan.hibernate-cache@13.0.6.Final//org.infinispan.hibernate.cache.v53.InfinispanRegionFactory.defineDataTypeCacheConfigurations(InfinispanRegionFactory.java:374)
	at org.infinispan.hibernate-cache@13.0.6.Final//org.infinispan.hibernate.cache.v53.InfinispanRegionFactory.start(InfinispanRegionFactory.java:242)
	... 24 more

```

In case it helps, my standalone.xml has:
```
            <cache-container name="hibernate" default-cache="local-query" modules="org.hibernate.infinispan">
                <local-cache name="entity">
                    <transaction mode="NON_XA"/>
                    <heap-memory size="10000"/>
                    <expiration max-idle="100000"/>
                </local-cache>
                <local-cache name="local-query">
                    <transaction mode="NONE"/>
                    <heap-memory size="10000"/>
                    <expiration max-idle="100000"/>
                </local-cache>
                <local-cache name="timestamps">
                    <transaction mode="NONE"/>
                </local-cache>
            </cache-container>
```

**Comment by scottmarlow** [comment]
> @scottmarlow Where did that config come from? The entity cache should not be transactional.

From our TCK testing.

The below is from latest WildFly Preview, should we update to the below for both WildFly + WildFly Preview?
```
            <cache-container name="hibernate" marshaller="JBOSS" modules="org.infinispan.hibernate-cache">
                <local-cache name="entity">
                    <heap-memory size="10000"/>
                    <expiration max-idle="100000"/>
                </local-cache>
                <local-cache name="local-query">
                    <heap-memory size="10000"/>
                    <expiration max-idle="100000"/>
                </local-cache>
                <local-cache name="timestamps">
                    <expiration interval="0"/>
                </local-cache>
                <local-cache name="pending-puts">
                    <expiration max-idle="60000"/>
                </local-cache>
            </cache-container>
```

**Comment by scottmarlow** [comment]
I will change the entity config to remove the transaction mode, weird that we didn't have to change that before. :-)

**Comment by pferraro** [comment]
@scottmarlow 
1. Yes - we should fix the TCK config.
1. You're right - we should not suddenly *have* to change it (Infinispan 2LC impl disables tx automatically).  It looks like invocation batching is getting inadvertently enabled in WF preview if the tx mode is NON_XA.  That should not be happening.  I'll figure out why and push a fix for that here.

**Comment by scottmarlow** [comment]
/me testing again locally with ^ updates...

**Comment by scottmarlow** [comment]
@pferraro `Test results: passed: 264`!

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**